src/subcommand/server.rs

```diff
@@ -1236,6 +1238,36 @@ impl Server {
     })
   }
 
+  async fn collections(
+    Extension(page_config): Extension<Arc<PageConfig>>,
+    Extension(index): Extension<Arc<Index>>,
+  ) -> ServerResult<Response> {
+    Self::collections_paginated(Extension(page_config), Extension(index), Path(0)).await
+  }
+
+  async fn collections_paginated(
+    Extension(page_config): Extension<Arc<PageConfig>>,
+    Extension(index): Extension<Arc<Index>>,
+    Path(page_index): Path<u64>,
+  ) -> ServerResult<Response> {
+    let (collections, more_collections) =
+      index.get_collections_paginated(100, page_index.try_into().unwrap())?;
```

Comment:
This needs to return an error if it fails, since otherwise on a 32 bit platform, the try into could fail and it could crash.